protocol/messenger_communities.go

```diff
@@ -2612,42 +2624,34 @@ func (m *Messenger) GetCommunityIDFromKey(communityKey string) string {
 	return communityID
 }
 
-// RequestCommunityInfoFromMailserver installs filter for community and requests its details
-// from mailserver. It waits until it has the community before returning it.
-// If useDatabase is true, it searches for community in database and does not request mailserver.
-func (m *Messenger) RequestCommunityInfoFromMailserver(privateOrPublicKey string, shard *common.Shard, useDatabase bool) (*communities.Community, error) {
-	communityID := m.GetCommunityIDFromKey(privateOrPublicKey)
-	if useDatabase {
+// FetchCommunity installs filter for community and requests its details
+// from mailserver.
+//
+// If `request.TryDatabase` is true, it first looks for community in database,
+// and requests from mailserver only if it wasn't found locally.
+// If `request.WaitForResponse` is true, it waits until it has the community before returning it.
+// If `request.WaitForResponse` is false, it installs filter for community and requests its details
+// from mailserver. When response received it will be passed through signals handler.
+func (m *Messenger) FetchCommunity(request *FetchCommunityRequest) (*communities.Community, error) {
```

Comment:
I think we could validate the request here before doing operations on it, since it comes from the user.
Make sure the communityID is not nul, if there is a chard, that the values are ok, stuff like that.